src/cryptography/hazmat/backends/openssl/decode_asn1.py

```diff
@@ -709,7 +709,11 @@ def _asn1_string_to_ascii(backend, asn1_string):
 def _asn1_string_to_utf8(backend, asn1_string):
     buf = backend._ffi.new("unsigned char **")
     res = backend._lib.ASN1_STRING_to_UTF8(buf, asn1_string)
-    backend.openssl_assert(res >= 0)
+    if res == -1:
+        raise TypeError(
```

Comment:
Should this be a `ValueError`? `TypeError` is usually on a "user called the function with the wrong type".